store/structured/mongo/store.go

```diff
@@ -45,21 +48,31 @@ func NewStore(p Params) (*Store, error) {
 		ctx:    context.Background(),
 	}
 
-	p.Lifecycle.Append(fx.Hook{
-		OnStart: func(ctx context.Context) error {
-			return store.Initialize(ctx)
-		},
-		OnStop: func(ctx context.Context) error {
-			return store.Terminate(ctx)
-		},
-	})
+	var err error
+
+	if p.Lifecycle != nil {
+		p.Lifecycle.Append(fx.Hook{
+			OnStart: func(ctx context.Context) error {
+				return store.Initialize(ctx)
+			},
+			OnStop: func(ctx context.Context) error {
+				return store.Terminate(ctx)
+			},
+		})
+	} else {
+		// If we're not using `fx`, then do the Initialization as part of `NewStore`
```

Comment:
I have always preferred to keep initialization logic outside of constructors (especially when networking is involved). I think the calling code should call `store.Initialize(ctx)` with its own context.